docs/reference/docs/delete-by-query.asciidoc

```diff
@@ -0,0 +1,311 @@
+[[docs-delete-by-query]]
+== Delete By Query API
+
+experimental[The delete-by-query API is new and should still be considered experimental.  The API may change in ways that are not backwards compatible]
+
+The simplest usage of `_delete_by_query` just performs a deletion on every
+document that match a query. Here is the API:
+
+[source,js]
+--------------------------------------------------
+DELETE /twitter/_delete_by_query
```

Comment:
I kind of wonder if `_delete_by_query` should be a `POST` instead of a `DELETE`. You aren't really deleting the thing at the URL so maybe `POST` is more right because you are causing the `_delete_by_query` action to be taken? I dunno.